modules/apps/callbacks/ibc_middleware_test.go

```diff
@@ -550,8 +550,21 @@ func (s *CallbacksTestSuite) TestOnTimeoutPacket() {
 				s.Require().Equal(1, sourceCounters[types.CallbackTypeSendPacket])
 				s.Require().Equal(uint8(2), sourceStatefulCounter)
 
+				// TODO: in order to unmarshal the transfer stack, we need to extract the transferstack
+				// from the legacy ibc module as the legacy ibc module UnmarshalPacketData is not what we need.
+				var unmarshaller porttypes.PacketDataUnmarshaler
+				if legacyModule, ok := cbs[0].(*porttypes.LegacyIBCModule); ok {
+					legacyModuleCbs := legacyModule.GetCallbacks()
+					// transfer stack is at index 0
+					unmarshaller, ok = legacyModuleCbs[0].(porttypes.PacketDataUnmarshaler)
+					s.Require().True(ok)
+				} else {
+					unmarshaller, ok = cbs[0].(porttypes.PacketDataUnmarshaler)
+					s.Require().True(ok)
+				}
```

Comment:
I believed this should be removed in https://github.com/cosmos/ibc-go/issues/7083

Maybe we can add a link in the comment.

Reply:
Done!